src/libopenrave/planningutils.cpp

```diff
@@ -4198,7 +4198,7 @@ int DynamicsCollisionConstraint::Check(const std::vector<dReal>& q0, const std::
                         BOOST_ASSERT(0);
                     } // end switch maskinterpolation
                     bool bFoundTimeInstant = false;
-                    dReal root;
```

Comment:
@Puttichai could you please provide a good default for this variable, because it's not assigned, still used for comparison within the loop (?), then conditionally assigned inside the loop and once again used after the loop (where it still may be non-initialized, because the loop was empty or the conditions were always false.

Reply:
`root` is used only when `bFoundTimeInstant` is `true`. `root` is assigned to a value at the same time `bFoundTimeInstant` is set to `true`. So I think it's fine.